diff --git a/example/ios/build.sh b/example/ios/build.sh
--- a/example/ios/build.sh
+++ b/example/ios/build.sh
@@ -21,15 +21,19 @@ set_cmake_options () {
   options="$options -DCMAKE_BUILD_TYPE=Release"
 }
 
-platforms="macOS iOS watchOS tvOS visionOS"
+platforms="$1"
+minimum_deployment_version="$2"
 #platforms="watchOS"
 for platform in $platforms;
 do
   echo "Platform = ${platform}"
-  if [[ $platform = "macOS" ]]; then
+  if [[ $platform == *"-simulator" ]]; then
+    platform="${platform%-simulator}"
+    simulators="1"
+  elif [[ $platform = "macOS" ]]; then
     simulators="0"
   else
-    simulators="0 1"
+    simulators="0"
   fi
 
   for simulator in $simulators;
@@ -57,7 +61,10 @@ do
       fi
 
       echo "iOS platform = ${ios_platform}"
       other_options="-DIOS_PLATFORM=${ios_platform} -DCMAKE_TOOLCHAIN_FILE=${td_path}/CMake/iOS.cmake"
+      if [[ -n "$minimum_deployment_version" ]]; then
+        other_options="$other_options -DIOS_DEPLOYMENT_TARGET=${minimum_deployment_version}"
+      fi
     fi
 
     set_cmake_options $platform
@@ -67,26 +74,6 @@ do
     cd $build
     cmake $td_path $options $other_options -DCMAKE_INSTALL_PREFIX=../${install}
     ZERO_AR_DATE=1 make -j3 install || exit
     cd ..
-    install_name_tool -id @rpath/libtdjson.dylib ${install}/lib/libtdjson.dylib
-    mkdir -p ../tdjson/${platform}/include
-    rsync --recursive ${install}/include/ ../tdjson/${platform}/include/
-    mkdir -p ../tdjson/${platform}/lib
-    cp ${install}/lib/libtdjson.dylib ../tdjson/${platform}/lib/
   done
 done
-
-produced_dylibs=(install-*/lib/libtdjson.dylib)
-xcodebuild_frameworks=()
-
-for dylib in "${produced_dylibs[@]}";
-do
-  xcodebuild_frameworks+=(-library $(grealpath "${dylib}"))
-done
-
-# Make xcframework
-xcodebuild -create-xcframework \
-    "${xcodebuild_frameworks[@]}" \
-    -output "libtdjson.xcframework"
-
-rsync --recursive libtdjson.xcframework ../tdjson/
