diff --git a/CMake/iOS.cmake b/CMake/iOS.cmake
index 1293040d4..ce1d09b97 100644
--- a/CMake/iOS.cmake
+++ b/CMake/iOS.cmake
@@ -195,13 +195,13 @@ set (CMAKE_OSX_SYSROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Sysroot used for iOS su
 # Set the architectures unless specified manually with IOS_ARCH
 if (NOT DEFINED IOS_ARCH)
     if (IOS_PLATFORM STREQUAL "OS")
-        set (IOS_ARCH "armv7;armv7s;arm64")
+        set (IOS_ARCH "arm64")
     elseif (IOS_PLATFORM STREQUAL "SIMULATOR")
-        set (IOS_ARCH "i386;x86_64;arm64")
+        set (IOS_ARCH "x86_64;arm64")
     elseif (IOS_PLATFORM STREQUAL "WATCHOS")
-        set (IOS_ARCH "armv7k;arm64_32;arm64")
+        set (IOS_ARCH "arm64_32;arm64")
     elseif (IOS_PLATFORM STREQUAL "WATCHSIMULATOR")
-        set (IOS_ARCH "i386;x86_64;arm64")
+        set (IOS_ARCH "x86_64;arm64")
     elseif (IOS_PLATFORM STREQUAL "TVOS")
         set (IOS_ARCH "arm64")
     elseif (IOS_PLATFORM STREQUAL "TVSIMULATOR")
