diff --git a/CMake/iOS.cmake b/CMake/iOS.cmake
index 935139542..1293040d4 100644
--- a/CMake/iOS.cmake
+++ b/CMake/iOS.cmake
@@ -199,7 +199,7 @@ if (NOT DEFINED IOS_ARCH)
     elseif (IOS_PLATFORM STREQUAL "SIMULATOR")
         set (IOS_ARCH "i386;x86_64;arm64")
     elseif (IOS_PLATFORM STREQUAL "WATCHOS")
-        set (IOS_ARCH "armv7k;arm64_32")
+        set (IOS_ARCH "armv7k;arm64_32;arm64")
     elseif (IOS_PLATFORM STREQUAL "WATCHSIMULATOR")
         set (IOS_ARCH "i386;x86_64;arm64")
     elseif (IOS_PLATFORM STREQUAL "TVOS")
